src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -447,27 +448,35 @@ trait Infer extends Checkable {
      *  @return map from tparams to inferred arg, if inference was successful, tparams that map to None are considered left undetermined
      *    type parameters that are inferred as `scala.Nothing` and that are not covariant in `restpe` are taken to be undetermined
      */
-    def adjustTypeArgs(tparams: List[Symbol], tvars: List[TypeVar], targs: List[Type], restpe: Type = WildcardType): AdjustedTypeArgs.Result  = {
-      val buf = AdjustedTypeArgs.Result.newBuilder[Symbol, Option[Type]]
+    def adjustTypeArgs(tparams: List[Symbol], tvars: List[TypeVar], targs: List[Type], restpe: Type = WildcardType): AdjustedTypeArgs  = {
+      val okParams = ListBuffer[Symbol]()
```

Comment:
After taking another look at this code, I realised that, much like your own suggestion in [another PR](https://github.com/scala/scala/pull/7875#issuecomment-473155072), it seems that here it holds that `tparams` is equal to `tvars map (_.origin.typeSymbol)`; from which follows that `okParams` here would be equal to `okArgs.map(_origin.typeSymbol)`. Perhaps we can avoid creating `okParams`, and propagate that to the users of this output? 

That can be done in another PR, though.